src/plugin.rs

```diff
@@ -50,12 +53,102 @@ impl Serialize for Error {
 
 type Result<T> = std::result::Result<T, Error>;
 
+/// Separates a "connection string" into it's constituant parts
+/// which are: (a) the database, and (b) the connection details.
+///
+/// In most cases you connect with network details but for a SQLite
+/// database it is with a file based resource where the file based resource
+/// will need to be combined with the app's designated directory in filesystem.
+enum Connection {
+    Sqlite(String, Option<PathBuf>),
+    Mysql(String),
+    Postgres(String),
+}
+
+impl Connection {
```

Comment:
The **Connection** enum allows us to keep the "app directory" stored to the side for when it's needed. The other DB's use network paths so this is not relevant.

Reply:
Ok interesting. I think part of me was just really in love with the idea of using Enums and Traits. I'll look into the cfg approach to understand it better. I do have two questions though:

1. I can see that you'd only want to compile the drivers for the DB's you are using but assuming that this is managed correctly, is there a notable sizing or performance issue with the current approach? 
2. Is this library not meant to be able to allow for multiple DB's being used? I think this would be an edge case but maybe not _that_ edge as I can see cases where SQLite is used as a local store but it interacts with Postgres remotely. Anyway, if this is the goal it feels like the cfg approach might be over-optimization but I'm perfectly fine to be wrong as my lens is quite cloudy.

Reply:
this enum has now been removed in favor of some of the conditional code blocks we discussed on Discord